sstables/mx/writer.cc

```diff
@@ -1385,7 +1385,9 @@ stop_iteration writer::consume_end_of_partition() {
 
     write(_sst.get_version(), *_data_writer, row_flags::end_of_partition);
 
-    if (_pi_write_m.promoted_index_size && _pi_write_m.first_clustering) {
+    bool is_first_partition = !_first_key;
+
+    if ((_pi_write_m.promoted_index_size || (is_first_partition && _cfg.force_promoted_index_for_first_partition)) && _pi_write_m.first_clustering) {
```

Comment:
I think we don't really need to promoted index - we can read the sstable from the beginning and stop on the first fragment. We could also read the last partition in reverse to find the last key.

Is that better or worse? We avoid playing tricks with configuring the writer, but we're causing more work. Maybe it's not better.

Reply:
To read last partition in reverse, we need promoted index anyway to find the offset of the last row, no?

Reply:
It will work whether we have a promoted index or not (just like a regular reversed read should work whether we have a promoted index or not).

Reply:
If we have 10000 files to process on boot, it means we read on worst case, about 63*1024 (size to not generate pi) * 10000 = ~615M for just getting the last row of the last partition. An average SSD is able to do that in less than 2 sec. A small price we're willing to pay, perhaps. I'll see how much extra work that is, compared to the promoted index approach implemented in this series. If it's much more work (I expect it not to be), then I'll stick with the current approach.